### PR TITLE
Add space around YAML editor at desktop

### PR DESCRIPTION
**Pull request by rhamilto**
Resolves https://jira.coreos.com/browse/CONSOLE-1693

localhost_9000_k8s_ns_robb_pods_etcd-operator-5dfbc9d746-m6jf9_yaml (1)
localhost_9000_k8s_ns_robb_pods_etcd-operator-5dfbc9d746-m6jf9_yaml (2)
localhost_9000_k8s_ns_robb_pods_etcd-operator-5dfbc9d746-m6jf9_yaml (3)
localhost_9000_k8s_ns_robb_pods_etcd-operator-5dfbc9d746-m6jf9_yaml
I opted to keep the bleed at mobile since space is so limited.
localhost_9000_k8s_ns_robb_pods_etcd-operator-5dfbc9d746-m6jf9_yaml (4)

### CODE REVIEW

File: frontend/public/components/edit-yaml.jsx [hunk 1]
```diff
@@ -159,7 +159,8 @@ export const EditYAML = connect(stateToProps)(
     }
 
     get editorHeight() {
-      return Math.floor(this.height - this.buttons.getBoundingClientRect().height);
+      // if viewport is > 767, also subtract top padding on .yaml-editor
```
Comment (rhamilto) [inline frontend/public/components/edit-yaml.jsx]: Not happy with the ternary and hard coded padding value here...

File: frontend/public/components/edit-yaml.jsx [hunk 2]
```diff
@@ -171,7 +172,8 @@ export const EditYAML = connect(stateToProps)(
 
     get width() {
       const sidebar = _.first(document.getElementsByClassName('co-p-has-sidebar__sidebar'));
-      const pageWidth = document.getElementById('content-scrollable').getBoundingClientRect().width;
+      // if viewport is > 767, also subtract left and right margins on .yaml-editor
```
Comment (rhamilto) [inline frontend/public/components/edit-yaml.jsx]: Not happy with the ternary and hard coded margin sum value here...

**Comment by rhamilto** [comment]
attn:  @spadgett, @serenamarie125, @bmignano, @alimobrem

**Comment by bmignano** [comment]
Looks good to me 👍 FYI @beanh66

File: frontend/public/components/edit-yaml.jsx [hunk 3]
```diff
@@ -170,11 +174,15 @@ export const EditYAML = connect(stateToProps)(
...
     get width() {
-      const sidebar = _.first(document.getElementsByClassName('co-p-has-sidebar__sidebar'));
-      const pageWidth = document.getElementById('content-scrollable').getBoundingClientRect().width;
-      return sidebar
-        ? pageWidth - sidebar.getBoundingClientRect().width
-        : pageWidth;
+      const hasSidebarSidebar = _.first(document.getElementsByClassName('co-p-has-sidebar__sidebar'));
+      const contentScrollableWidth = document.getElementById('content-scrollable').getBoundingClientRect().width;
+      // if viewport width is > 767, also subtract left and right margins on .yaml-editor
+      const hasSidebarBodyWidth = window.innerWidth > 767
```
Comment (spadgett) [inline frontend/public/components/edit-yaml.jsx]: Consider const vars for the breakpoint and margin since each are repeated

**Comment by rhamilto** [comment]
Thank you for the helpful feedback, @spadgett.  Updated.

**Pull request merged**